File: dsa_leetcode_beginner/0226-invert-binary-tree/0226-invert-binary-tree.js
class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

function invertTree(root) {
    if (root === null) {
        return null;
    }

    // Swap the left and right children
    [root.left, root.right] = [root.right, root.left];

    // Recursively invert the left and right subtree
    invertTree(root.left);
    invertTree(root.right);

    return root;
}

File: dsa_leetcode_beginner/0226-invert-binary-tree/0226-invert-binary-tree.cpp
#include <iostream>

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr) {
            return nullptr;
        }

        // Swap the left and right children
        TreeNode* temp = root->left;
        root->left = root->right;
        root->right = temp;

        // Recursively invert the left and right subtree
        invertTree(root->left);
        invertTree(root->right);

        return root;
    }
};